pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -17,151 +17,191 @@ limitations under the License.
-	actualPods := podManager.GetPods()
-	if len(actualPods) != len(expectedPods) {
-		t.Errorf("expected %d pods, got %d pods; expected pods %#v, got pods %#v", len(expectedPods), len(actualPods),
-			expectedPods, actualPods)
-	}
-	for _, expected := range expectedPods {
-		found := false
-		for _, actual := range actualPods {
-			if actual.UID == expected.UID {
-				if !reflect.DeepEqual(&expected, &actual) {
-					t.Errorf("pod was recorded incorrectly. expect: %#v, got: %#v", expected, actual)
-				}
-				found = true
-				break
-			}
 		}
-		if !found {
-			t.Errorf("pod %q was not found in %#v", expected.UID, actualPods)
+		normalPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("normal-pod-uid"),
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
 		}
+	)
+
+	testCase := []struct {
+		name         string
+		podList      []*v1.Pod
+		wantPod      *v1.Pod
+		expectPods   []*v1.Pod
+		expectGetPod *v1.Pod
+		expectUID    types.UID
+	}{
+		{
+			name:         "Get normal pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      normalPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: normalPod,
+			expectUID:    types.UID("normal-pod-uid"),
+		},
+		{
+			name:         "Get static pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      staticPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: staticPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
+		{
+			name:         "Get mirror pod",
+			podList:      []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantPod:      mirrorPod,
+			expectPods:   []*v1.Pod{staticPod, normalPod},
+			expectGetPod: staticPod,
+			expectUID:    types.UID("static-pod-uid"),
+		},
 	}
-	// Tests UID translation works as expected. Convert static pod UID for comparison only.
-	if uid := podManager.TranslatePodUID(mirrorPod.UID); uid != kubetypes.ResolvedPodUID(staticPod.UID) {
-		t.Errorf("unable to translate UID %q to the static POD's UID %q; %#v",
-			mirrorPod.UID, staticPod.UID, podManager.mirrorPodByUID)
-	}
-
-	// Test the basic Get methods.
-	actualPod, ok := podManager.GetPodByFullName("bar_default")
-	if !ok || !reflect.DeepEqual(actualPod, staticPod) {
-		t.Errorf("unable to get pod by full name; expected: %#v, got: %#v", staticPod, actualPod)
-	}
-	actualPod, ok = podManager.GetPodByName("default", "bar")
-	if !ok || !reflect.DeepEqual(actualPod, staticPod) {
-		t.Errorf("unable to get pod by name; expected: %#v, got: %#v", staticPod, actualPod)
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager := newTestManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+
+			sortOpt := cmpopts.SortSlices(func(a, b *v1.Pod) bool { return a.Name < b.Name })
+			assert.Empty(t, cmp.Diff(actualPods, test.expectPods, sortOpt), "actualPods and expectPods are not equal")
```

Comment:
Please, change the argument order here and below, so that diff would look more natural, e.g.
```suggestion
			assert.Empty(t, cmp.Diff(test.expectPods, actualPods, sortOpt), "expected and actual pods differ")
```

Reply:
@googs1025 Please, address this comment, thanks!